### DISCUSSION TITLE
下書き記事の画面遷移なしの更新機能

### DISCUSSION THREAD

**Discussion by hirota-fs** [increments/qiita-discussions#364] [要望・アイデア・不具合]
## 困っていること
記事執筆中に誤操作でタブを削除してしまい、記事が書き直しになったことが度々ありました。
こうならない様に下書き記事の画面遷移なしのセーブ機能が欲しいです。

## 実現して欲しいこと
* 下書きを更新後、引き続き記事の編集ができること（画面遷移しないこと）
* 上記が繰り返し行えること
* 従来通り下書き編集画面から任意のタイミングで記事の公開ができること

**Comment by kyntk** [comment]
@hirota-fs
ご意見ありがとうございます。

実は下書きにはすでに自動保存の機能があります！
今回提案いただいた、「実現して欲しいこと」の3つはすでにできるようになっています。

今回のようにタブ削除の後タブを復元したときに、一見保存されていないように見えることがあるのですが、保存されている可能性があります。
次回は下書きが残っているか、下書き一覧をご確認いただけますでしょうか:bow:
https://qiita.com/drafts

ただし、タブ削除後の挙動が分かりづらいのは間違いありません。
運営としても課題に感じているため改善予定です :bow:

**Comment by hirota-fs** [comment]
@kyntk 
ご返答ありがとうございます。

実はタブ削除後すぐに確認していたのですが、完全にはタブ削除前の状態には戻りませんでした。
その後再度書き直しも行ったので確実だと思います。
部分的には自動保存機能が働いていたのかも知れませんが、更新内容を全て把握していなかったので
どの程度機能していたかは確認できておりません。

ですので自動保存機能も可能であれば、タイプが止まった瞬間など頻繁に更新を確認するように
改善していただけますと大変ありがたいです。

**Reply by kyntk** [reply]
@hirota-fs 

> タイプが止まった瞬間など頻繁に更新を確認するように

現在もタイプが止まるごとに1秒程度で保存する設定となっています。
しかし、ネットワーク等の問題で下書き保存に失敗する可能性もあります。
エディタ左下の「最終保存」の時刻から、下書きを最終保存した時刻が分かるので、こちらを確認して下書きが保存されているかを確認いただけますでしょうか。 :pray:

<img width="461" alt="image" src="https://user-images.githubusercontent.com/17251829/204198452-05c16cef-5000-4ca5-b21b-2587bf5982bb.png">

**Reply by hirota-fs** [reply]
@kyntk 
さきほど確認してみたところ
* タイプが止まった時点で保存されていること
* ならびにタブを削除した際に変更内容が保存されていること

以上2点が確認できました。
最初に投稿させていただいた不調はネットワークの不調に起因するものである確率が高そうです。
ご丁寧にありがとうございました。
